device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -1166,6 +1166,10 @@ private void acknowledgeReceivedMessage(IotHubTransportMessage receivedMessage)
                     {
                         CorrelatingMessageCallback callback = correlationCallbacks.get(correlationId);
 
+                        // We need to remove the CorrelatingMessageCallback with the current correlation ID from the map after the received C2D
+                        // message has been acknowledged. Otherwise, the size of map will grow endlessly which results in OutOfMemory eventually.
+                        correlationCallbacks.remove(correlationId);
```

Comment:
Should we wait until after the ack callback has been invoked to remove this from the list, i.e. after L1176?